src/cloudai/schema/system/slurm/slurm_system.py

```diff
@@ -49,7 +49,7 @@ class SlurmSystem(System):
     """
 
     @classmethod
-    def parse_node_list(cls, node_list: List[str]) -> List[str]:
+    def parse_node_list(cls, node_list: str) -> List[str]:
```

Comment:
Please add unit tests just for this function:
```py
@pytest.mark.parametrize(...)
def test_parse_node_list(input: str, expected: List[str]):
    ...
```

So far I find its code very confusing and not following the documentation of the function.

Reply:
how is it confusing ? It's just parsing the outputs of sinfo.
Anyways, I just fixed the function, the description and its use are the same as before.

I'll add a unittest

Reply:
Function arg is defined as a string: `node_list: str`.
But documentation says it should be a list of strings. So I wonder how is that even possible that input is different, but usage wasn't updated. I hope unit-tests will clarify it.

Reply:
ah yes this part is my mistake it used to be list[str] now its str, I changed the arg but forgot to change the description
I'm updating now